From fcdc77986120ef2b3eb30b802984d2c7fc53c2af Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 28 Jan 2019 17:59:37 +0100
Subject: [PATCH 0920/1051] arm64: dts: 9130: reduce the spi max frequency

The on-board mux does not allow higher freq than 400k - reduce it.

Change-Id: I5d5af2660987e441b3833860c7c4d8df4628d151
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/2962
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-db-A.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
index 7000bfacf22d..d4652c6c012b 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
@@ -104,7 +104,8 @@
 		#size-cells = <0x1>;
 		compatible = "jedec,spi-nor";
 		reg = <0x0>;
-		spi-max-frequency = <108000000>;
+		/* On-board MUX does not allow higher frequencies */
+		spi-max-frequency = <40000000>;
 
 		partitions {
 			compatible = "fixed-partitions";
-- 
2.17.1

